src/hypertable.c

```diff
@@ -3014,3 +3020,125 @@ ts_hypertable_update_dimension_partitions(const Hypertable *ht)
+			 quote_identifier(NameStr(ht->fd.schema_name)),
+			 quote_identifier(NameStr(ht->fd.table_name)));
+
+	int32 dimension_slice_id = ts_chunk_get_osm_slice_id(osm_chunk_id, time_dim->fd.id);
+
+	/*
+	 * range_start, range_end arguments must be converted to internal representation
+	 * a NULL start value is interpreted as INT64_MAX - 1 and a NULL end value is
+	 * interpreted as infinity.
+	 * Passing both start and end NULL values will reset the range to the default range an
+	 * OSM chunk is given upon creation, which is [INT64_MAX - 1, INT64_MAX]
+	 */
+	int64 range_start_internal, range_end_internal;
+	if (PG_ARGISNULL(1))
+		range_start_internal = PG_INT64_MAX - 1;
+	else
+		range_start_internal = ts_time_value_to_internal(PG_GETARG_DATUM(1), time_type);
+	if (PG_ARGISNULL(2))
+		range_end_internal = PG_INT64_MAX;
+	else
+		range_end_internal = ts_time_value_to_internal(PG_GETARG_DATUM(2), time_type);
+
+	if (range_start_internal > range_end_internal)
+		ereport(ERROR, errmsg("dimension slice range_end cannot be less than range_start"));
+
+	bool osm_chunk_empty = PG_GETARG_BOOL(3);
+
+	bool overlap = false, range_invalid = false;
+
+	ScanTupLock tuplock = {
+		.lockmode = LockTupleExclusive,
+		.waitpolicy = LockWaitBlock,
+	};
+	DimensionSlice *slice =
+		ts_dimension_slice_scan_by_id_and_lock(dimension_slice_id, &tuplock, CurrentMemoryContext);
+
+	if (!slice)
+		ereport(ERROR, errmsg("could not find slice with id %d", dimension_slice_id));
+	overlap = ts_osm_chunk_range_overlaps(dimension_slice_id,
+										  slice->fd.dimension_id,
+										  range_start_internal,
+										  range_end_internal);
+	/*
+	 * It should not be possible for OSM chunks to overlap with the range
+	 * managed by timescaledb. OSM extension should update the range of the
+	 * OSM chunk to [INT64_MAX -1, infinity) when it detects that it is
+	 * noncontiguous, so we should not end up detecting overlaps anyway.
+	 * But throw an error in case we encounter this situation.
+	 */
+	if (overlap)
+		ereport(ERROR,
+				errmsg("attempting to set overlapping range for tiered chunk of %s.%s",
+					   NameStr(ht->fd.schema_name),
+					   NameStr(ht->fd.table_name)),
+				errhint("Range should be set to invalid for tiered chunk"));
+	range_invalid = ts_osm_chunk_range_is_invalid(range_start_internal, range_end_internal);
+	/* Update the hypertable flags regarding the validity of the OSM range */
+	if (overlap || range_invalid)
+	{
+		if (!osm_chunk_empty)
+			ht->fd.status =
+				ts_set_flags_32(ht->fd.status, HYPERTABLE_STATUS_OSM_CHUNK_NONCONTIGUOUS);
+		else
+		{
+			/* range is also set to infinity so the OSM chunk is considered last */
+			range_start_internal = PG_INT64_MAX - 1;
+			range_end_internal = PG_INT64_MAX;
+			ht->fd.status =
+				ts_clear_flags_32(ht->fd.status, HYPERTABLE_STATUS_OSM_CHUNK_NONCONTIGUOUS);
+		}
+	}
+	else
+		ht->fd.status = ts_clear_flags_32(ht->fd.status, HYPERTABLE_STATUS_OSM_CHUNK_NONCONTIGUOUS);
```

Comment:
overlap is an error now so we dont need it as part of if, maybe we can also disentangle the nested ifs a bit and flatten it

Reply:
done